frame/merkle-mountain-range/src/lib.rs

```diff
@@ -214,10 +223,19 @@ pub mod pallet {
 			let data = T::LeafData::leaf_data();
 			// append new leaf to MMR
 			let mut mmr: ModuleMmr<mmr::storage::RuntimeStorage, T, I> = mmr::Mmr::new(leaves);
-			mmr.push(data).expect("MMR push never fails.");
-
-			// update the size
-			let (leaves, root) = mmr.finalize().expect("MMR finalize never fails.");
+			// MMR push never fails.
+			let _ = mmr.push(data);
```

Comment:
Given the change just below, I'm assuming this change's intent is to strip the pallet of possible panics via `expect()` calls. Will still get an error logged via https://github.com/paritytech/substrate/blob/2bededbca1b593ee60ccd290c0a21898087c5f13/frame/merkle-mountain-range/src/mmr/mmr.rs#L139
but might become fragile to depend on that remaining the same, and on 
https://github.com/paritytech/substrate/blob/2bededbca1b593ee60ccd290c0a21898087c5f13/frame/merkle-mountain-range/src/lib.rs#L244-L246 to later return `T::WeightInfo::on_initialize(peaks_before)`.

So at the price of verbosity, minor nit is to bake in both via
```suggestion
			if mmr.push(data).is_none() {
				frame_support::log::error!(
					target: "runtime::mmr", "MMR push failed"
				);
				return T::WeightInfo::on_initialize(peaks_before)
			}
```

Reply:
Right, the main point is to rid the pallet of any potential panic points. Panicking in runtime code breaks the chain - any node trying to run the block panics - so any `expect/unwrap` should be limited to case where we are 1000% sure it will never panic. Which is not the case here, if we change underlying function or mmr-lib or whatever we introduce panic possibility (even if tests should catch it, I'd rather we just remove the possibility altogether).